lib/chef/provider/package/aix.rb

```diff
@@ -83,21 +96,14 @@ def load_current_resource
         end
 
         def candidate_version
-          return @candidate_version if @candidate_version
-          ret = shell_out_with_timeout("installp -L -d #{@new_resource.source}")
-          ret.stdout.each_line do |line|
-            case line
-            when /\w:#{Regexp.escape(@new_resource.package_name)}:(.*)/
-              fields = line.split(":")
-              @candidate_version = fields[2]
-              @new_resource.version(fields[2])
-              Chef::Log.debug("#{@new_resource} setting install candidate version to #{@candidate_version}")
+          @candidate_version ||= begin
+            candidate, fs = filter_source_device
+            if candidate
+              @candidate_version = candidate
+              new_resource.version(candidate)
+              Chef::Log.debug("#{new_resource} setting install candidate version to #{@candidate_version}")
```

Comment:
I don't think this is ever setting `@candidate_version`, I think you're getting mostly the expected behavior because of `new_resource.version(candidate)` but line 102 where `@candidate_version = candidate` is going to get overwritten with whatever `Chef::Log.debug` returns.

```
Is this really setting `@candidate_version`? I see youirb(main):025:0> @foo
=> nil
irb(main):026:0> @foo ||= begin
irb(main):027:1* @foo = "the candidate version"
irb(main):028:1> puts "this is a debug statement"
irb(main):029:1> end
this is a debug statement
=> nil
irb(main):030:0> @foo
=> nil
```